sdk/core/azure-core/CMakeLists.txt

```diff
@@ -148,6 +148,13 @@ if(BUILD_TRANSPORT_WINHTTP)
   target_link_libraries(azure-core PRIVATE winhttp)
 endif()
 
+if (MSVC)
+  # Disable warnings:
+  # - C6285: (<non-zero constant> || <non-zero constant>) -> VBProject static analysis on Functional header:
+  #          _Is_large, regression from VS version 19.28.29915.0 to 19.29.30037.0
+  target_compile_options(azure-core PUBLIC /wd6285)
```

Comment:
Is there a bug we can track for the fix, i.e. when can we turn this back on?

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/2354

Reply:
Presumably the change that triggered this is https://github.com/microsoft/STL/pull/698? The documentation suggests this warning should never fire for `<boolean constant expression> || <boolean constant expression>`, so I can't see this as anything other than a compiler bug. Please file a bug report with a minimal repro on [Developer Community](https://developercommunity.visualstudio.com/report?space=62&entry=problem).

Reply:
https://developercommunity.visualstudio.com/t/MSVC-static-analysis-fails-due-to-_Is_l/1434886?space=62&entry=myfeedback

Reply:
see also: https://developercommunity.visualstudio.com/t/functional813-has-warning-c6285/1433378